pkg/tsdb/cloudwatch/log_actions.go

```diff
@@ -0,0 +1,318 @@
+	for _, logGroup := range response.LogGroups {
+		logGroupNames = append(logGroupNames, logGroup.LogGroupName)
+	}
+
+	groupNamesField := data.NewField("logGroupName", nil, logGroupNames)
+	frame := data.NewFrame("logGroups", groupNamesField)
+
+	return frame, nil
+}
+
+func (e *CloudWatchExecutor) executeStartQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, timeRange *tsdb.TimeRange) (*cloudwatchlogs.StartQueryOutput, error) {
+	startTime, err := timeRange.ParseFrom()
+	if err != nil {
+		return nil, err
+	}
+
+	endTime, err := timeRange.ParseTo()
+	if err != nil {
+		return nil, err
+	}
+
+	if !startTime.Before(endTime) {
+		return nil, fmt.Errorf("Invalid time range: Start time must be before end time")
+	}
+
+	startQueryInput := &cloudwatchlogs.StartQueryInput{
+		StartTime:     aws.Int64(startTime.Unix()),
+		EndTime:       aws.Int64(endTime.Unix()),
+		Limit:         aws.Int64(parameters.Get("limit").MustInt64(1000)),
+		LogGroupNames: aws.StringSlice(parameters.Get("logGroupNames").MustStringArray()),
+		QueryString:   aws.String("fields @timestamp,@log,@logStream|" + parameters.Get("queryString").MustString("")),
+	}
+	return logsClient.StartQueryWithContext(ctx, startQueryInput)
+}
+
+func (e *CloudWatchExecutor) handleStartQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, timeRange *tsdb.TimeRange, refID string) (*data.Frame, error) {
+	startQueryResponse, err := e.executeStartQuery(ctx, logsClient, parameters, timeRange)
+	if err != nil {
+		return nil, fmt.Errorf(err.(awserr.Error).Message())
+	}
+
+	dataFrame := data.NewFrame(refID, data.NewField("queryId", nil, []string{*startQueryResponse.QueryId}))
+	dataFrame.RefID = refID
+
+	clientRegion := parameters.Get("region").MustString("default")
+
+	dataFrame.Meta = &data.FrameMeta{
+		Custom: map[string]interface{}{
+			"Region": clientRegion,
+		},
+	}
+
+	return dataFrame, nil
+}
+
+func (e *CloudWatchExecutor) executeStopQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*cloudwatchlogs.StopQueryOutput, error) {
+	queryInput := &cloudwatchlogs.StopQueryInput{
+		QueryId: aws.String(parameters.Get("queryId").MustString()),
+	}
+
+	response, err := logsClient.StopQueryWithContext(ctx, queryInput)
+	if err != nil {
+		awsErr, _ := err.(awserr.Error)
+		if awsErr.Code() == "InvalidParameterException" {
+			response = &cloudwatchlogs.StopQueryOutput{Success: aws.Bool(false)}
+			err = nil
+		} else {
+			err = fmt.Errorf(err.(awserr.Error).Message())
+		}
+	}
+
+	return response, err
+}
+
+func (e *CloudWatchExecutor) handleStopQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*data.Frame, error) {
+	response, err := e.executeStopQuery(ctx, logsClient, parameters)
+
+	if err != nil {
+		return nil, fmt.Errorf(err.(awserr.Error).Message())
+	}
```

Comment:
What's the benefit of converting the error instead of returning the original one?
```suggestion
	response, err := e.executeStopQuery(ctx, logsClient, parameters)
	if err != nil {
		return nil, fmt.Errorf(err.(awserr.Error).Message())
	}
```